plugins/org.komodo.core/src/org/komodo/repository/RepositoryImpl.java

```diff
@@ -1243,4 +1254,18 @@ public void unpublish( final UnitOfWork uow,
 
     }
 
+    @Override
+    public KomodoObject komodoRoot(final UnitOfWork uow) throws KException {
+        return create(uow, KOMODO_ROOT, KomodoLexicon.Komodo.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoLibrary(final UnitOfWork uow) throws KException {
+        return create(uow, LIBRARY_ROOT, KomodoLexicon.Library.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoWorkspace(final UnitOfWork uow) throws KException {
```

Comment:
Should this method and the `komodoLibrary` method first call `komodoRoot` to makes sure the root is setup with the proper name and type?